content/d3/concepts/interactivity/interactivity.md

```diff
@@ -0,0 +1,55 @@
+---
+Title: 'Interactivity'
+Description: 'D3 interactivity enriches visualizations with dynamic user engagement and interactive features.'
+Subjects:
+  - 'Code Foundations'
+  - 'Data Visualization'
+Tags:
+  - 'D3'
+  - 'Data'
+CatalogContent:
+  - 'interactive-data-visualization-with-generative-ai'
+  - 'paths/data-visualization'
+---
+
+D3.js, or Data-Driven Documents, is a powerful JavaScript library for creating dynamic and interactive data visualizations in web browsers. It facilitates the binding of data to DOM elements, enabling seamless updates and transitions. With its declarative approach to manipulating the Document Object Model (DOM), D3.js empowers developers to create engaging and responsive visualizations, making it a go-to tool for building data-driven applications on the web.
+
+## Adding Tooltips for Enhanced Information
+
+Tooltips are a crucial element in interactive visualizations, providing additional context when users hover over data points. The following code demonstrates how to integrate tooltips into your D3.js visualization. This is a very basic example, it creates a square block. If you hover over the square block, you will see a tooltip under the block saying "this is a square!"
+
+```js
+ // Select the square element
+        const square = d3.select("rect.square");
+
+        // Event listener for mouseover event
+        square.on("mouseover", () => {
+            // Show the tooltip
+            d3.select(".tooltip")
+                .style("opacity", 1)
+                .style("left", d3.event.pageX + "px")
+                .style("top", d3.event.pageY + "px");
+        });
+
+        // Event listener for mouseout event
+        square.on("mouseout", () => {
+            // Hide the tooltip
+            d3.select(".tooltip").style("opacity", 0);
+        });
+```
+
+Make sure you have your html file setup using this script before running.
+
+## Output
+
+Block when mouse pointer is not hovering:
+![Screenshot 2023-12-19 014458](https://github.com/ShatilKhan/docs/assets/52494840/6f444ebe-bc6b-4526-b365-412735c854a5)
+
+Tooltip appears when hovered over:
+![Screenshot 2023-12-19 014419](https://github.com/ShatilKhan/docs/assets/52494840/8c69038e-c4d2-498c-9dfa-f58839e21cfb)
```

Comment:
Take reference from our [style guide](https://github.com/Codecademy/docs/blob/main/documentation/style-guide.md) on how to add an image/media file to the content.

Reply:
Am I supposed to commit directly to the main branch?
The style guide says images should be stored on main/tree/media of the main docs repo

Reply:
Follow the same method as you've followed till now. Have a look at this -
- If you'd like to host files such as images or GIFs, please upload them to the [/media folder](https://github.com/Codecademy/docs/tree/main/media).
- Images should be linked to the URL: "https://raw.githubusercontent.com/Codecademy/docs/main/media/" followed by the image name.
- File names should use dashes (-) to separate words rather than spaces or underscores.
- File names should be in lowercase.